src/ngraph/builder/reduce_ops.cpp

```diff
@@ -111,5 +112,68 @@ namespace ngraph
             return result->add_provenance_group_members_above({value});
         }
 
+        std::shared_ptr<Node> builder::opset1::mean(const Output<Node>& value,
+                                                    const AxisSet& reduction_axes)
+        {
+            const auto xsum = std::make_shared<ngraph::opset1::ReduceSum>(
+                value,
+                ngraph::opset1::Constant::create(
+                    element::i64, Shape{reduction_axes.size()}, reduction_axes.to_vector()),
+                false);
+
+            const auto N = get_num_elements(value.get_shape(), reduction_axes);
```

Comment:
We may rewrite the `get_num_elements` function in a dynamic way: Just use `v1::Gather` to collect dims from `reduction_axes` and then `ReduceProd`. I'd suggest also to check whether `reduction_axes` size is equal to passed shape size. In that situation this would be sth like this:
```
ReduceProd(value_shape, Range(0, ShapeOf(value_shape), Const(1)))
```

Reply:
It will be implemented in task #1070